File: ecmascript/bob/bob.js
class Bob {
  hey(message) {
    message = message.trim();
    if (message.length == 0) {
      return 'Fine. Be that way!';
    } else if (message == message.toUpperCase() && message != message.toLowerCase()) {
      return 'Whoa, chill out!';
    } else if (message.endsWith('?')) {
      return 'Sure.';
    } else {
      return 'Whatever.';
    }
  }
}

export default Bob;

File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn- exclamation? [input]
  (and
   (= input (str/upper-case input))
   (not (= input (str/lower-case input)))))

(defn- silence? [input]
  (str/blank? input))

(defn- question? [input]
  (= \? (last input)))

(defn response-for [input]
  (cond
    (exclamation? input) "Whoa, chill out!"
    (silence? input) "Fine. Be that way!"
    (question? input) "Sure."
    :else "Whatever."))
